src/main/java/com/iexec/sms/tee/session/palaemon/PalaemonSessionService.java

```diff
@@ -248,18 +248,22 @@ public String getSessionYml(PalaemonSessionRequest request) throws Exception {
                                 secretIndex)
                         .map(TeeTaskComputeSecret::getValue)
                         .orElse(EMPTY_YML_VALUE);
-        tokens.put(IEXEC_APP_DEVELOPER_SECRET_PREFIX + secretIndex, appDeveloperSecret0);
+        tokens.put(IexecEnvUtils.IEXEC_APP_DEVELOPER_SECRET_PREFIX + secretIndex, appDeveloperSecret0);
 
-        String requesterSecret0 =
-                teeTaskComputeSecretService.getSecret(
-                                OnChainObjectType.APPLICATION,
-                                taskDescription.getAppAddress(),
-                                SecretOwnerRole.REQUESTER,
-                                taskDescription.getRequester(),
-                                secretIndex)
-                        .map(TeeTaskComputeSecret::getValue)
-                        .orElse(EMPTY_YML_VALUE);
-        tokens.put(IEXEC_REQUESTER_SECRET_PREFIX + secretIndex, requesterSecret0);
+        HashMap<String, String> requesterSecrets = new HashMap<>();
+        for (Map.Entry<String, String> secretEntry: taskDescription.getSecrets().entrySet()) {
+            String requesterSecret =
```

Comment:
To be discussed, requester secrets leak should not exceed `maxRequesterSecretCount`, e.g:
```
for(i = 0; i < maxRequesterSecretCount){
   requesterSecretKey = taskDescription.getSecrets().get(i);
   if(requesterSecretKey){
      requesterSecret = teeTaskComputeSecretService.getSecret(requesterSecretKey);
      requesterSecrets.put(IexecEnvUtils.IEXEC_REQUESTER_SECRET_PREFIX + i, requesterSecret);
   }
}
```

Otherwise It looks we don't need `maxRequesterSecretCount`.

Reply:
done in 2d0d899

Reply:
Thanks!